### DISCUSSION TITLE
Difference in dimension between LIS and WRF grid

### DISCUSSION THREAD

**Discussion by shaider2** [NASA-LIS/LISF#1665] [General]
Hi all,
I am using LIS as an offline model. I want to replace soil moisture of wrfinput by LIS output. However, when I was replacing the variable there was a difference in dimension between the variables. I had used similar domain and resolution for both models. 
It will be great if anyone can provide any suggestions regarding this.

**Comment by dmocko** [comment]
Hi @shaider2 

You may wish to start using our NASA Unified - WRF system, which makes it easy to use LIS output as WRF initialization:
https://nuwrf.gsfc.nasa.gov/

Our LIS output soil moisture (in netCDF format) is typically written with these dimensions (using "ncdump" to view the file):
```
        float SoilMoist_inst(SoilMoist_profiles, north_south, east_west) ;
                SoilMoist_inst:units = "m^3 m-3" ;
                SoilMoist_inst:standard_name = "soil_moisture_content" ;
                SoilMoist_inst:long_name = "soil moisture content" ;
                SoilMoist_inst:scale_factor = 1.f ;
                SoilMoist_inst:add_offset = 0.f ;
                SoilMoist_inst:missing_value = -9999.f ;
                SoilMoist_inst:_FillValue = -9999.f ;
                SoilMoist_inst:vmin = 0.f ;
                SoilMoist_inst:vmax = 0.f ;
```
where:
```
dimensions:
        east_west = 464 ;
        north_south = 224 ;
        time = UNLIMITED ; // (1 currently)
        SoilMoist_profiles = 4 ;
```

I'm not sure what dimensions are expected in wrfinput.  Can you please share?

**Reply by dmocko** [reply]
My apologies, but I am not familiar at all with wrfinput or using WRF.

I have asked some team members with more familiarity on WRF to assist on your question.

Please be patient, as many folks are taking time off for the holidays and end of the year.

**Comment by shaider2** [comment]
Hi @dmocko ,
Thank you so much for the reply. For my simulations, I have used similar domain for LIS and WRF (i.e. reference latitude, reference longitude and number of grid is same). I have attached the ldt config file and namelist.wps for your reference. 
However, the number of dimension for soil moisture in LIS and WRF are different. I have also attached the latitude and longitude value for both models. 
[lat_lon.xlsx](https://github.com/user-attachments/files/18271249/lat_lon.xlsx)
[ldt.config.prelis.docx](https://github.com/user-attachments/files/18271250/ldt.config.prelis.docx)
[namelist.wps.docx](https://github.com/user-attachments/files/18271251/namelist.wps.docx)


**Variable: SMOIS_lis**
Type: float
Total Size: 1928784 bytes
            482196 values
Number of Dimensions: 3
Dimensions and sizes:   [SoilMoist_profiles | 4] x [north_south | 281] x [east_west | 429]
Coordinates:
Number Of Attributes: 9
  units :       m^3 m-3
  standard_name :       soil_moisture_content
  long_name :   soil moisture content
  scale_factor :         1
  add_offset :   0
  missing_value :       -9999
  _FillValue :  -9999
  vmin :        -1e+15
  vmax :        1e+15


**Variable: SMOIS_wrf**
Type: float
Total Size: 1917440 bytes
            479360 values
Number of Dimensions: 4
Dimensions and sizes:   [Time | 1] x [soil_layers_stag | 4] x [south_north | 280] x [west_east | 428]
Coordinates:
Number Of Attributes: 6
  FieldType :   104
  MemoryOrder : XYZ
  description : SOIL MOISTURE
  units :       m3 m-3
  stagger :     Z
  coordinates : XLONG XLAT XTIME

**Reply by jinwoongyoo** [reply]
Hi @shaider2 
Please make sure your WRF domain grid sizes for Lat and Lon are legitimate.
Your settings for the dims are [Time | 1] x [soil_layers_stag | 4] x [south_north | 280] x [west_east | 428].

However, if your parent-child domain grid ratio is 1:3 as usual,  south_north and west_east should be integers of 3y+1 and 3x+1, where y and x are certain integers for south_north and west_east dimensions, respectively.
Now your 280 = 3x93+1 and 428 = 3x142+2.
So your WRF WPS will crash regardless of your LIS input.

Therefore, first fix your WRF domain grid dimension sizes to be legitimate like 280 and 430 for example.
Then, set your LIS domain grid sizes to be 1 grid less than your WRF domain grids such as 279 and 429 in this case.
Please give it a try and let us know.

Thank you.

Regards,
Jinwoong
